src/cms/models/users/user.py

```diff
@@ -0,0 +1,240 @@
+        """
+        Return all unread messages of this user
+
+        :return: The unread messages of this user
+        :rtype: ~django.db.models.query.QuerySet [ ~cms.models.chat.chat_message.ChatMessage ]
+        """
+        return ChatMessage.history.filter(sent_datetime__gt=self.chat_last_visited)
+
+    def update_chat_last_visited(self):
+        """
+        Update the :attr:`~cms.models.users.user.User.chat_last_visited` to the current time
+
+        :return: Return the previous :attr:`~cms.models.users.user.User.chat_last_visited` value
+        :rtype: ~datetime.datetime
+        """
+        previous_chat_last_visited = self.chat_last_visited
+        self.chat_last_visited = timezone.now()
+        self.save()
+        logger.debug(
+            "Field chat_last_visited of %r updated from %s to %s",
+            self,
+            previous_chat_last_visited.strftime("%Y-%m-%d %H:%M:%S"),
+            self.chat_last_visited.strftime("%Y-%m-%d %H:%M:%S"),
+        )
+        return previous_chat_last_visited
+
+    def __str__(self):
+        """
+        This overwrites the default Django :meth:`~django.db.models.Model.__str__` method which would return ``User object (id)``.
+        It is used in the Django admin backend and as label for ModelChoiceFields.
+
+        :return: A readable string representation of the user
+        :rtype: str
+        """
+        return self.full_user_name
+
+    def __repr__(self):
+        """
+        This overwrites the default Django ``__repr__()`` method which would return ``<User: User object (id)>``.
+        It is used for logging.
+
+        :return: The canonical string representation of the user
+        :rtype: str
+        """
+        optional_fields = ""
+        if not self._state.adding:
+            if self.is_staff:  # pylint: disable=W0125
+                if self.role:
+                    optional_fields += f", team: {self.role.english_name}"
+                if self.is_superuser:
+                    optional_fields += ", superuser"
+                else:
+                    optional_fields += ", staff"
+            else:
+                if self.role:
+                    optional_fields += f", role: {self.role.english_name}"
+                if self.regions.count() == 1:
+                    optional_fields += f", region: {self.regions.first().name}"
+        return f"<User (id: {self.id}, username: {self.username}{optional_fields})>"
+
+    class Meta:
+        #: The verbose name of the model
+        verbose_name = _("user")
+        #: The plural verbose name of the model
+        verbose_name_plural = _("users")
+        #: The default permissions for this model
+        default_permissions = ()
+        #: Persmissions for the model (default)
+        permissions = [
+            ("add_user", "Can add user"),
+            ("change_user", "Can change user"),
+            ("del_user", "Can delete user"),
+            ("view_user", "Can view user"),
+        ]
```

Comment:
```suggestion
        default_permissions = ("change", "delete", "view")
```

Since the custom permissions defined here are nearly identical with the default permissions, I'd go with the default ones. The only difference is that instead of `del_user`, the permissions `delete_user` would have to be checked (which is more consistent with all our other permissions)... and a quick search in the source code revealed that you used `@permission_required("cms.delete_user")` in the code anyway, so `del_user` doesn't seem to be used in permission checks.

Also, permissions are sometimes checked via `perms.auth`, these would have to be replaced with `perms.cms`...